nixos/modules/programs/benchexec.nix

```diff
@@ -0,0 +1,95 @@
+{ lib
+, pkgs
+, config
+, options
+, ...
+}:
+let
+  cfg = config.programs.benchexec;
+  opt = options.programs.benchexec;
+
+  # NOTE: `x` is asserted to be of type 'string' or 'int'
+  # by `options.programs.benchexec.users.type` (see below).
+  # NOTE: If `x` is a string, existence of `config.users.users.${x}.uid`
+  # is asserted by `config.assertions` (see below).
+  # However, it is possible that this function is evaluated on a
+  # nonexistent user before the assertion is evaluated, so
+  # the case of a nonexistent user still does not throw an error,
+  # but instead uses 'nobody'.
+  uid = x: if builtins.isString x then (config.users.users.${x} or config.users.users.nobody).uid else x;
+in
+{
+  options.programs.benchexec = {
+    enable = lib.mkEnableOption "BenchExec";
+    package = lib.options.mkPackageOption pkgs "benchexec" { };
+
+    users = lib.options.mkOption {
+      type = with lib.types; listOf (either str int);
+      description = lib.mdDoc ''
+        Users that intend to use BenchExec.
+        Provide usernames of users that are configured via {option}`${options.users.users}` as string,
+        and UIDs of "mutable users" as integers.
+        Control group delegation will be configured via systemd.
+        For more information, see <https://github.com/sosy-lab/benchexec/blob/3.18/doc/INSTALL.md#setting-up-cgroups>.
+      '';
+      default = [ ];
+      example = lib.literalExpression ''
+        [
+          "alice" # username of a user configured via ${options.users.users}
+          1007    # UID of a mutable user
+        ]
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = (map
+      (user: {
+        assertion = config.users.users ? ${user};
+        message = ''
+          The user '${user}' intends to use BenchExec (via `${opt.users}`), but is not configured via `${options.users.users}`.
+        '';
+      })
+      (builtins.filter builtins.isString cfg.users)
+    ) ++ (map
+      (id: {
+        assertion = config.users.mutableUsers;
+        message = ''
+          The user with UID '${id}' intends to use BenchExec (via `${opt.users}`), but mutable users are disabled via `${options.users.mutableUsers}`.
+        '';
+      })
+      (builtins.filter builtins.isInt cfg.users)
+    );
+
+    environment.systemPackages = [ cfg.package ];
+
+    systemd = {
+      enableUnifiedCgroupHierarchy = true;
```

Comment:
This should already be the case, isn't it?

Reply:
Converted to an assertion in 8a4cc816e0ff